src/storage/v2/disk/storage.cpp

```diff
@@ -915,6 +926,12 @@ void DiskStorage::DiskAccessor::PrefetchOutEdges(const VertexAccessor &vertex_ac
         }
       }
     }
+    for (auto &deleted_edge_key : edges_to_delete_) {
+      if (deleted_edge_key == diskEdgeKey) {
+        // We already deleted this edge.
+        return false;
+      }
+    }
```

Comment:
Hmm here we have to iterate over all deleted edges which is expensive. I think a better solution would be to check whether edge_accessor is alrerady deleted.

Reply:
while iterating over edge_acc. Same for InEdges

Reply:
Don't you think that iterating through all edge_acc would be more expensive?